core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java

```diff
@@ -57,32 +59,18 @@ public void testGeoPointReset() throws IOException {
     }
 
     public void testEqualsHashCodeContract() {
-        // generate a random geopoint
-        final GeoPoint x = RandomGeoGenerator.randomPoint(random());
-        final GeoPoint y = new GeoPoint(x.lat(), x.lon());
-        final GeoPoint z = new GeoPoint(y.lat(), y.lon());
-        // GeoPoint doesn't care about coordinate system bounds, this simply validates inequality
-        final GeoPoint a = new GeoPoint(x.lat() + randomIntBetween(1, 5), x.lon() + randomIntBetween(1, 5));
-
-        /** equality test */
-        // reflexive
-        assertTrue(x.equals(x));
-        // symmetry
-        assertTrue(x.equals(y));
-        // transitivity
-        assertTrue(y.equals(z));
-        assertTrue(x.equals(z));
-        // inequality
-        assertFalse(x.equals(a));
-
-        /** hashCode test */
-        // symmetry
-        assertTrue(x.hashCode() == y.hashCode());
-        // transitivity
-        assertTrue(y.hashCode() == z.hashCode());
-        assertTrue(x.hashCode() == z.hashCode());
-        // inequality
-        assertFalse(x.hashCode() == a.hashCode());
+        // GeoPoint doesn't care about coordinate system bounds, this simply validates equality and hashCode.
+        final DoubleSupplier randomDelta = () -> randomValueOtherThan(0.0, () -> randomDoubleBetween(-1000000, 1000000, true));
+        checkEqualsAndHashCode(RandomGeoGenerator.randomPoint(random()), GeoPoint::new, pt -> {
+            switch (randomInt(2)) {
```

Comment:
I think this is an unnecessary use of randomization in a test. I think randomizing in situations like this only makes tests harder to debug. Why not simply test all the cases, we can enumerate them?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/27634/commits/8109cafd77d9830eeca465261f30c3158abc4137.